app/services/burn-event/service.go

```diff
@@ -0,0 +1,178 @@
+			return
+		}
+		transactionResponse, err = s.hederaNodeClient.
+			SubmitScheduledTokenTransferTransaction(event.Amount, tokenID, event.Recipient, s.thresholdAccount, s.payerAccount, event.TxHash)
+	}
+	if err != nil {
+		s.logger.Errorf("[%s] - Failed to submit scheduled transaction. Error [%s].", event.TxHash, err)
+		return
+	}
+
+	s.logger.Infof("[%s] - Successfully submitted scheduled transaction [%s] for [%s] to receive [%d] tinybars.",
+		event.TxHash,
+		transactionResponse.TransactionID, event.Recipient, event.Amount)
+
+	txReceipt, err := transactionResponse.GetReceipt(s.hederaNodeClient.GetClient())
+	if err != nil {
+		s.logger.Errorf("[%s] - Failed to get transaction receipt for [%s]", event.TxHash, transactionResponse.TransactionID)
+		return
+	}
+
+	switch txReceipt.Status {
+	case hedera.StatusIdenticalScheduleAlreadyCreated:
+		s.handleScheduleSign(event.TxHash, helper.ToMirrorNodeTransactionID(txReceipt.ScheduledTransactionID.String()), *txReceipt.ScheduleID)
+	case hedera.StatusSuccess:
+		transactionID := helper.ToMirrorNodeTransactionID(txReceipt.ScheduledTransactionID.String())
+		s.logger.Infof("[%s] - Updating db status to Submitted with TransactionID [%s].", event.TxHash, transactionID)
+
+		err := s.repository.UpdateStatusSubmitted(event.TxHash, txReceipt.ScheduleID.String(), transactionID)
+		if err != nil {
+			s.logger.Errorf(
+				"[%s] - Failed to update submitted status with TransactionID [%s], ScheduleID [%s]. Error [%s].",
+				event.TxHash, transactionID, txReceipt.ScheduleID, err)
+			return
+		}
+	default:
+		s.logger.Errorf("[%s] - TX [%s] - Scheduled Transaction resolved with [%s].", event.TxHash, transactionResponse.TransactionID, txReceipt.Status)
+
+		err := s.repository.UpdateStatusFailed(transactionResponse.TransactionID.String())
+		if err != nil {
+			s.logger.Errorf("[%s] - Failed to update status failed. Error [%s].", transactionResponse.TransactionID.String(), err)
+			return
+		}
+		return
+	}
+
+	transactionID := helper.ToMirrorNodeTransactionID(txReceipt.ScheduledTransactionID.String())
+
+	onSuccess, onFail := s.scheduledTxExecutionCallbacks(transactionID)
+	s.mirrorNodeClient.WaitForScheduledTransferTransaction(transactionID, onSuccess, onFail)
+}
+
+func (s *Service) handleScheduleSign(txHash, transactionID string, scheduleID hedera.ScheduleID) {
+	s.logger.Debugf("[%s] - Scheduled transaction already created - Executing Scheduled Sign for [%s].", txHash, scheduleID)
+	txResponse, err := s.hederaNodeClient.SubmitScheduleSign(scheduleID)
+	if err != nil {
+		s.logger.Errorf("[%s] - Failed to submit schedule sign [%s]. Error: [%s].", txHash, scheduleID, err)
+		return
+	}
+
+	receipt, err := txResponse.GetReceipt(s.hederaNodeClient.GetClient())
+	if err != nil {
+		s.logger.Errorf("[%s] - Failed to get transaction receipt for schedule sign [%s]. Error: [%s].", txHash, scheduleID, err)
+		return
+	}
+
+	if receipt.Status != hedera.StatusSuccess {
```

Comment:
I think that we should explicitly check the status of the receipt against `already executed`. 
If already executed -> log specific message with debug/trace that this TX ended up one of the last ones and the tx already executed.
If the status is != success and `alredy execute`, then we should log an error similar to what you are logging.